### ISSUE TITLE
Allow supplementary groups to be defined for startup script

### ISSUE THREAD

**Issue by adammcdonagh** [sensu/sensu-go#3699]
On RHEL6 where systemd is not yet being used, chroot is being used to start up the agent.

This is fine in most cases, however on systems where the sensu user has been added to additional groups, e.g so it can read config files owned by another application (as part of a check), this causes issues.

When running `id` under the agent's environment, the following is returned:

`uid=4444(sensu) gid=4444(sensu) groups=4444(sensu),0(root)`

This does not correctly reflect all the groups the user is in.

## Feature Suggestion
Make use of the `/etc/default/sensu-agent` which is referenced in the init script, allowing you to specify supplementary groups to be passed to the chroot command via the `--groups` option

## Possible Implementation
```bash
# cat /etc/default/sensu-agent
groups=mygroup

# Add to sensu-agent init script:

[ ! -z "$groups" ] && groups="--groups $groups"

# Change chroot command:
chroot --userspec "$user":"$group" $groups "$chroot" sh -c "
```

When running id via the agent, you then get:
`uid=4444(sensu) gid=4444(sensu) groups=4444(sensu),27202(mygroup)`

## Context
In many instances we might have a monitor where Sensu needs to open a file owned by a different application. Rather than adding sudo entries and running the whole script as root or a different user, it's simpler to simply add the sensu user to an appropriate OS group so that it can open the files. This works fine when testing scripts in the shell, however due to the way the agent is started, it doesn't work when running as a daemon.

I've tested this on RHEL 7 using systemd, it doesn't seem to have the same behaviour and inherits the secondary groups just fine.

In the meantime, I'll have to modify the startup scripts on my RHEL 5 systems

**Comment by adammcdonagh** [comment]
I'm assuming the packaging scripts (whatever you use to create the RPMs) are held in a different repo somewhere? I'd offer to contribute a PR if they were accessible.

**Comment by amdprophet** [comment]
Your assumption is correct; our packaging is in a different (private) repository at the moment. Your request makes sense to me and I think we should be able to add it to our packaging scripts.

Thank you for providing a possible implementation! We'll likely use something along the lines of `SENSU_CHROOT_ADDITIONAL_GROUPS` for the environment variable to avoid conflicts & make the intent of the variable clear. I'll update this issue once we're ready to release the feature.

**Comment by amdprophet** [comment]
A fix for this went out with the 6.0.0 release back in August. Groups can be set by setting the `groups` environment variable in `/etc/default/sensu-[agent|backend]` or `/etc/sysconfig/sensu-[agent|backend]`. Groups should be comma-delimited without spaces (e.g. `groups="group1,group2,group3"`).
